Src/PerformanceCollector/Shared/Implementation/QuickPulse/IQuickPulseServiceClient.cs

```diff
@@ -16,15 +17,33 @@ internal interface IQuickPulseServiceClient
         /// </summary>
         /// <param name="instrumentationKey">InstrumentationKey for which to submit data samples.</param>
         /// <param name="timestamp">Timestamp to pass to the server.</param>
+        /// <param name="configurationETag">Current configuration ETag that the client has.</param>
+        /// <param name="authApiKey">Authentication API key.</param>
+        /// <param name="configurationInfo">When available, the deserialized response data received from the server.</param>
         /// <returns><b>true</b> if data is expected, otherwise <b>false</b>.</returns>
-        bool? Ping(string instrumentationKey, DateTimeOffset timestamp);
+        bool? Ping(
```

Comment:
Well.
The reason why QuickPulseServiceClient exists is to communicate with QPS. QPS supports two major operations - PING and POST. For POST, I call it SubmitSamples in the interface (since POST is confusing on many levels :) ). For PING, I just use Ping, because that's essentially what it does - informs the service that we're alive.
Of course, it also does many other things, like passing local timestamp and identifying information, etc. It now also supplies configuration etag and reads an updated configuration back if present in the response. Still, it's the QPS's PING operation, and I think it makes sense to call it that.
It never made any particular sense to me to create a new instance of this class more than once. In fact, a lot of the stuff that it uses never or almost never changes, so I should probably make it part of its internal state. I think I'll do it in the next iteration since parameter lists do become a bit long.